meilisearch-http/src/index_controller/update_actor/handle_impl.rs

```diff
@@ -0,0 +1,96 @@
+    UpdateMsg, UpdateStatus,
+};
+use crate::index_controller::IndexActorHandle;
+
+#[derive(Clone)]
+pub struct UpdateActorHandleImpl<D> {
+    sender: mpsc::Sender<UpdateMsg<D>>,
+}
+
+impl<D> UpdateActorHandleImpl<D>
+where
+    D: AsRef<[u8]> + Sized + 'static + Sync + Send,
+{
+    pub fn new<I>(
+        index_handle: I,
+        path: impl AsRef<Path>,
+        update_store_size: usize,
+    ) -> anyhow::Result<Self>
+    where
+        I: IndexActorHandle + Clone + Send + Sync + 'static,
+    {
+        let path = path.as_ref().to_owned().join("updates");
+        let (sender, receiver) = mpsc::channel(100);
+        let store = MapUpdateStoreStore::new(index_handle.clone(), &path, update_store_size);
+        let actor = UpdateActor::new(store, receiver, path, index_handle)?;
+
+        tokio::task::spawn(actor.run());
+
+        Ok(Self { sender })
+    }
+}
+#[async_trait::async_trait]
+impl<D> UpdateActorHandle for UpdateActorHandleImpl<D>
+where
+    D: AsRef<[u8]> + Sized + 'static + Sync + Send,
+{
+    type Data = D;
+
+    async fn update(
+        &self,
+        meta: UpdateMeta,
+        data: mpsc::Receiver<PayloadData<Self::Data>>,
+        uuid: Uuid,
+    ) -> Result<UpdateStatus> {
+        let (ret, receiver) = oneshot::channel();
+        let msg = UpdateMsg::Update {
+            uuid,
+            data,
+            meta,
+            ret,
+        };
+        let _ = self.sender.send(msg).await;
+        receiver.await.expect("update actor killed.")
+    }
+    async fn get_all_updates_status(&self, uuid: Uuid) -> Result<Vec<UpdateStatus>> {
+        let (ret, receiver) = oneshot::channel();
+        let msg = UpdateMsg::ListUpdates { uuid, ret };
+        let _ = self.sender.send(msg).await;
+        receiver.await.expect("update actor killed.")
+    }
+
+    async fn update_status(&self, uuid: Uuid, id: u64) -> Result<UpdateStatus> {
+        let (ret, receiver) = oneshot::channel();
+        let msg = UpdateMsg::GetUpdate { uuid, id, ret };
+        let _ = self.sender.send(msg).await;
+        receiver.await.expect("update actor killed.")
+    }
+
+    async fn delete(&self, uuid: Uuid) -> Result<()> {
+        let (ret, receiver) = oneshot::channel();
+        let msg = UpdateMsg::Delete { uuid, ret };
+        let _ = self.sender.send(msg).await;
+        receiver.await.expect("update actor killed.")
+    }
+
+    async fn create(&self, uuid: Uuid) -> Result<()> {
+        let (ret, receiver) = oneshot::channel();
+        let msg = UpdateMsg::Create { uuid, ret };
+        let _ = self.sender.send(msg).await;
+        receiver.await.expect("update actor killed.")
+    }
+
+    async fn snapshot(&self, uuid: Uuid, path: PathBuf) -> Result<()> {
+        let (ret, receiver) = oneshot::channel();
+        let msg = UpdateMsg::Snapshot { uuid, path, ret };
+        let _ = self.sender.send(msg).await;
+        receiver.await.expect("update actor killed.")
+    }
+}
```

Comment:
Do you think it would be interesting to write a macro that generates all these functions? (for another PR though, this one is already complex enough :joy: )

Reply:
Yes I thought about that a bit, it's not straightforward, but could be a fun little project :)

Reply:
@irevoire lol I tried to do so and it looks pretty strange and I'm not sure it became any better  😄 
https://github.com/meilisearch/transplant/commit/3e875b05c93736ecfe6cfea8014627bae6f394b9

It works, however!